src/tools/prefs.win.cpp

```diff
@@ -99,10 +100,18 @@ static bool save_prefs(fs::path fpath) {
 static bool load_prefs(fs::path fpath) {
 	prefsDirty = false;
 	std::map<std::string,boost::any> temp_prefs;
-	std::ifstream fin(fpath.string().c_str());
+	std::istream* in;
```

Comment:
Please don't use raw pointers. This should probably be an `std::unique_ptr`.

Reply:
Should I also change the method signature of pop_next_action() to return `unique_ptr<Element>`  instead of `Element*`? Seems like that would be good, because I realized I'm cloning Elements and passing the clones out of pop_next_action() but not deleting them ever.

Reply:
Something like that is definitely needed. I'm not sure if that exact thing is it. It all depends on how `Clone()` works.